src/parts/value.class.ts

```diff
@@ -0,0 +1,53 @@
+import {
+    OperationHistory,
+    OperationHistoryType,
+} from "./operation-history.interface";
+
+export class Value implements OperationHistory {
+    static readonly ZERO: Value = new Value(0);
+
+    constructor(public readonly value: number) {}
+
+    public get type() {
+        return OperationHistoryType.VALUE;
+    }
+
+    get length(): number {
+        return this.toString().length;
+    }
+
+    isZero(): Boolean {
+        return this.value === 0;
+    }
+
+    append(other: Value): Value {
+        return new Value(this.value * Math.pow(10, other.length) + other.value);
```

Comment:
This wouldn't work if the value has a fractional part?
e.g. (10.3333).append (33)